src/views/ProductView.vue

```diff
@@ -578,4 +622,247 @@ function goToWishlists() {
         name: 'Wishlists',
     })
 }
+
+/**
+ * Reference to trigger the getInventoryStatusOfProductItems query.
+ */
+const triggerGetInventoryStatusOfProductItemsQuery = ref(0)
+
+/**
+ * Asynchronously computes the inventory status of product items.
+ */
+const getInventoryStatusOfProductItemsQuery = asyncComputed(
+    async () => {
+        triggerGetInventoryStatusOfProductItemsQuery.value
+
+        if (activeUserRoleIsEitherAdminOrEmployee.value) {
+            if (productVariantId.value !== undefined) {
+                return client.getInventoryStatusOfProductItems({
+                    productVariantId: productVariantId.value,
+                })
+            }
+        }
+
+        return null
+    },
+    null,
+    {
+        onError: (e) =>
+            pushErrorNotification(
+                errorMessages.getInventoryStatusOfProductItems,
+                e
+            ),
+        shallow: false,
+    }
+)
+
+/**
+ * Computed property for the inventory status of product items.
+ */
+const inventoryStatusOfProductItems = computed(() => {
+    if (getInventoryStatusOfProductItemsQuery.value !== null) {
+        const nodes =
+            getInventoryStatusOfProductItemsQuery.value
+                .productItemsByProductVariant.nodes
+        if (nodes !== null && nodes !== undefined) {
+            return nodes
+        }
+    }
+
+    return null
+})
+
+/**
+ * Computed property indicating whether there are product items in stock.
+ * @returns True if there are product items in stock, false otherwise.
+ */
+const inStock = computed(() => {
```

Comment:
I know this is currently the only way how you can do this.
However this is also really inefficient, in particular with lots of product items.
Imho the `productItems` field should provide a `filter` option, so that you can just use `totalCount` to fetch this information.
Maybe open an issue on `inventory`?

Reply:
I agree! I will open an issue.

Reply:
[The issue](https://frontend.gropius.duckdns.org/components/ba383e91-738a-488f-baf4-4b5d8876bd74/issues/41d14a98-238e-42ad-b72c-14b6aa313beb)